nixd/lib/Controller/Hover.cpp

```diff
@@ -22,6 +22,31 @@ using namespace lspserver;
 
 namespace {
 
+class OptionsHoverProvider {
+  AttrSetClient &Client;
+
+public:
+  OptionsHoverProvider(AttrSetClient &Client) : Client(Client) {}
+  std::optional<OptionDescription> resolveHover(std::vector<std::string> Scope,
```

Comment:
Pass this vector in const-reference, for better performance.

```suggestion
  std::optional<OptionDescription> resolveHover(const std::vector<std::string> &Scope,
```